deploy/crds/integreatly.org_grafanadashboards_crd.yaml

```diff
@@ -0,0 +1,99 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: grafanadashboards.integreatly.org
+spec:
+  group: integreatly.org
+  names:
+    kind: GrafanaDashboard
+    listKind: GrafanaDashboardList
+    plural: grafanadashboards
+    singular: grafanadashboard
+  scope: ""
+  validation:
+    openAPIV3Schema:
+      description: GrafanaDashboard is the Schema for the grafanadashboards API
+      properties:
+        apiVersion:
+          description: 'APIVersion defines the versioned schema of this representation
+            of an object. Servers should convert recognized schemas to the latest
+            internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+          type: string
+        kind:
+          description: 'Kind is a string value representing the REST resource this
+            object represents. Servers may infer this from the endpoint the client
+            submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+          type: string
+        metadata:
+          type: object
+        spec:
+          description: GrafanaDashboardSpec defines the desired state of GrafanaDashboard
+          properties:
+            dashboard:
+              description: GrafanaDashboardSpec defines the desired state of GrafanaDashboard
+              properties:
+                json:
+                  description: 'INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+                    Important: Run "operator-sdk generate k8s" to regenerate code
+                    after modifying this file'
+                  type: string
+                name:
+                  type: string
+                plugins:
+                  items:
+                    description: GrafanaPlugin contains information about a single
+                      plugin
+                    properties:
+                      name:
+                        type: string
+                      version:
+                        type: string
+                    required:
+                    - name
+                    - version
+                    type: object
+                  type: array
+                url:
+                  type: string
+              required:
+              - json
+              - name
+              type: object
+            grafanaRef:
```

Comment:
So adding this field, and nesting the dashboard content under the new `dashboard` field makes this a backwards incpomatible change, so we need to consider bumping the major to 3 @pb82. (There are similar changes to the other 2 CRDs as well.

@lleszczu If I understand correctly, the `grafanaRef` field is to assist with checking if a dashboard has already been loaded into the grafana API.
Is there are reason not to use the existing `spec.name` field instead?

Reply:
@david-martin I'd also prefer to keep this backwards compatible if possible. One way to achieve this would be to support `spec.json` field as a deprecated property and give preference to `spec.dashboard` if provided. On the other hand, proper versioning of CRDs seems something inevitable in a longer term anyway.

@lleszczu I wonder, do we even need to check if a dashboard has already been imported? Would the API not simply ignore it if it already exists? Or does this require the dashboard to have an UID?

Reply:
@david-martin
I had to break API for datasources as they currently allow to define many datasources with single CRD, which is not very grafana api friendly,  altought v1alpha1 means that those definitions are not stable and users should expect changes, v1alpha2 could be solution for that, but it would require maintaining support for versioning and I'm not sure how to handle that at this moment.
Basically that breaking change is the reason why I bumped operator version to 2.1.
Regarding grafanaRef, its purpose is to reference which grafana should be used for this resource, so you can have 2 (or more 😄 ) grafanas with different API address, different credentials, etc. I imagine that this operator was not exactly meant to support this usecase, but I believe that for development and testing of whole monitoring solution this may be useful ( for example monitoring-stable and monitoring-playground namespaces on the same k8s cluster).
@pb82 
Currently operator just checks `.status` of the resource and if it states the resource is correctly provisioned and not changed it's not touching grafana API.

Also regarding adding this .spec.dashboard placeholder, I believe it gives nice consistency across different CRDs that may be added in future (grafana API supports provisioning organizations, users, folders, etc.)